src/probnum/diffeq/_probsolve_ivp.py

```diff
@@ -275,11 +275,14 @@ def probsolve_ivp(
     measmod = odefiltsmooth.GaussianIVPFilter.string_to_measurement_model(
         method, ivp, prior_process
     )
-    solver = odefiltsmooth.GaussianIVPFilter.construct_with_rk_init(
-        ivp,
-        prior_process,
-        measmod,
+
+    rk_init = odefiltsmooth.initialize.RungeKuttaInitialization()
+    solver = odefiltsmooth.GaussianIVPFilter(
+        ivp=ivp,
+        prior_process=prior_process,
+        measurement_model=measmod,
         with_smoothing=dense_output,
+        initialization_routine=rk_init,
```

Comment:
In my opinion, passing the initialisation routine with these objects is so easy that the alternative constructors are rather in the way than helpful.